### PR TITLE
Added WebsocketBindingWithMembers

### PR DESCRIPTION
**Pull request by AlexejStukov**
WebsocketBindingWithMembers inherits WebsocketBinding and additionally enables sending of member variables, properties and methods.

### CODE REVIEW

**Comment by AlexejStukov** [comment]
Might also be integrated in ``Binding`` or as a Mixin.

File: channels/binding/websockets.py [hunk 1]
```diff
@@ -87,3 +87,42 @@ def update(self, pk, data):
...
+    """
+
+    send_members = []
+
+    def serialize_data(self, instance):
+        data = super(WebsocketBindingWithMembers, self).serialize_data(instance)
+        for m in self.send_members:
+            member = getattr(instance, m)
+            try:
+                data[m] = member()
```
Comment (andrewgodwin) [inline channels/binding/websockets.py]: I suggest using `if callable(member)` here instead?

File: channels/binding/websockets.py [hunk 2]
```diff
@@ -87,3 +87,42 @@ def update(self, pk, data):
...
+            except TypeError:
+                data[m] = member
+        return data
```
Comment (andrewgodwin) [inline channels/binding/websockets.py]: Maybe consider catching/warning about types that won't serialize into JSON; for the fields, this is done automatically with the Django serializer.

**Comment by AlexejStukov** [comment]
Wouldn't  the encoding step already raise a ``TypeError: XYZ is not JSON serializable``?

**Comment by andrewgodwin** [comment]
Well the Django code specifically encodes a few extra types using known patterns, so I guess we could use that, but not required.

Also, absolute -1 to mixins. I will not merge with one :)

**Comment by AlexejStukov** [comment]
Hm, ok; then undo the change. Why do you dislike them?

**Comment by andrewgodwin** [comment]
They make it very hard to debug where things come from and are generally very unfriendly to newer Python programmers, so I tend to avoid them at all costs. I've had to maintain too many codebases that rely on mixins to work and it's impossible to work out where anything happens.

**Comment by AlexejStukov** [comment]
Now it's implemented as a normal class again.

**Comment by andrewgodwin** [comment]
Yup, nice.

**Pull request merged**